Added docs for running GUI tests (#62)
owncloud/docs-client-desktop: modules/ROOT/pages/appendices/guitest.adoc

:free-trial-url: https://www.froglogic.com/squish/free-trial/
:pythonchanger-url: https://kb.froglogic.com/squish/howto/using-external-python-interpreter-squish-6-6/PythonChanger.py
:owncloud-test-middleware-url: https://github.com/owncloud/owncloud-test-middleware
:test-case-scenario-url: https://bdd.tips/#chapter=9
:squish-docker-image-url: https://hub.docker.com/r/owncloudci/squish
:acceptance-tests-url: https://doc.owncloud.com/server/developer_manual/testing/acceptance-tests.html#how-to-write-acceptance-tests
:stepdefinitions-url: https://github.com/owncloud/owncloud-test-middleware/tree/main/src/stepDefinitions
:object-map-object-url: https://www.froglogic.com/squish/features/object-map-object-identification-tools/
:client-repo-url: https://github.com/owncloud/client/

== Introduction

This document explains how to run GUI tests for the Desktop Client locally in your system. To run GUI tests, the Squish GUI Test Automation Tool for all kinds of cross-platform desktop, mobile, embedded and web applications is used.

== Prerequisites

Before we can actually run the test, we will need to make a Desktop Client build and install and configure Squish first.

> NOTE: Before running tests, you need to make sure that you have disabled `oauth2` and `openidconnect` in owncloud server.

Reviewer: ```suggestion
NOTE: Before running tests, you need to make sure that you have disabled `oauth2` and `openidconnect` in ownCloud Server.
```

=== Building the Desktop Client

To be able to run tests, you need to xref:appendices/building.adoc[build the Desktop Client] from latest master.

=== Install Squish

> After building the ownCloud Client, install and configure squish. There are two ways to install squish:

Reviewer: ```suggestion
After building the ownCloud Client, install and configure squish. There are two ways to install Squish:
```

* xref:installing-squish-using-an-executable-file[Installing Squish Using an Executable File]
* xref:installing-squish-using-docker[Installing Squish Using Docker]

==== Installing Squish Using an Executable File

To install Squish, follow these steps:

. Download the latest version of Squish from {squish-url}[froglogic] to a location of your choice.
. Depending upon the version and system you are using, you will get a file like `squish-6.6.2-qt512x-linux64.run`.

> . Make the downloaded file executable by runninng following example command:

Reviewer 2: ```suggestion
. Make the downloaded file executable by running the following example command:
```

+
[source,console]
----
sudo chmod +x ./squish-6.6.2-qt512x-linux64.run
----
. Execute the file:
+
[source,console]
----
sudo ./squish-6.6.2-qt512x-linux64.run
----
. You will be asked for a license key. When asked, enter your existing license or get a {free-trial-url}[free trial license]
. After you have entered the license key, Squish opens.

==== Installing Squish Using Docker

> You can also use the {squish-docker-image-url}[squish docker image] to run tests. Just proceed with following steps:

Reviewer: ```suggestion
You can also use the {squish-docker-image-url}[Squish docker image] to run tests. Proceed with the following steps:
```

> Pull the docker image with following command:

Reviewer: ```suggestion
Pull the docker image with the following command:
```

[source,console]
----
sudo docker pull owncloudci/squish
----

=== Configure Squish

After installing Squish, follow these steps to configure it:

. Close Squish if opened.
. pythonchanger-url[Download] the `PythonChanger.py` script and save it in your squish installation folder.
. Run the downloaded script.
+
[source,console]
----
sudo python3 PythonChanger.py --force
----

== Prepare Tests

> Some necessary steps before running test.

Reviewer: ```suggestion
Some necessary steps before running tests:
```

. Clone the {client-repo-url}[Desktop Client] from GitHub
. Copy `test/gui/config.sample.ini` to `test/suite_oc-desktop/config.ini`

> . Edit `test/gui/config.ini` and set `BACKEND_HOST=` to the URL of your owncloud server e.g. `BACKEND_HOST=http://localhost/owncloud-core`

Reviewer: ```suggestion
. Edit `test/gui/config.ini` and set `BACKEND_HOST=` to the URL of your ownCloud server, e.g. `BACKEND_HOST=http://localhost/owncloud-core`
```

. Start Squish
. Open the existing test-suite via: menu:File[Open Test Suite > test/gui]
. Go to menu:Edit[Server Settings > Manage AUTs > Mapped AUTs]

> . Click btn:[Add] and select the compiled binary, eg: `client/client-build/bin/owncloud`

Reviewer: ```suggestion
. Click btn:[Add] and select the compiled binary, e.g.: `client/client-build/bin/owncloud`
```

> . Close any running Desktop Client, by clicking btn:[Quit ownCloud] in the settings page

Reviewer: ```suggestion
. Close any running Desktop Clients by clicking btn:[Quit ownCloud] in the settings page
```

. Run the AUT (Application under Test) to check if everything works properly via menu:Run[Lauch AUT] (the Desktop Client settings window should appear)

> == Run Rests

Reviewer: ```suggestion
== Run Tests
```

:squish-docker-image-url: https://hub.docker.com/r/owncloudci/squish
:acceptance-tests-url: https://doc.owncloud.com/server/developer_manual/testing/acceptance-tests.html#how-to-write-acceptance-tests
:stepdefinitions-url: https://github.com/owncloud/owncloud-test-middleware/tree/main/src/stepDefinitions
:object-map-object-url: https://www.froglogic.com/squish/features/object-map-object-identification-tools/
:client-repo-url: https://github.com/owncloud/client/

== Introduction

This document explains how to run GUI tests for the Desktop Client locally in your system. To run GUI tests, the Squish GUI Test Automation Tool for all kinds of cross-platform desktop, mobile, embedded and web applications is used.

== Prerequisites

Before we can actually run the test, we will need to make a Desktop Client build and install and configure Squish first.

NOTE: Before running tests, you need to make sure that you have disabled `oauth2` and `openidconnect` in ownCloud Server.

=== Building the Desktop Client

To be able to run tests, you need to xref:appendices/building.adoc[build the Desktop Client] from latest master.

=== Install Squish

> After building the ownCloud Client, install and configure squish.

Reviewer: ```suggestion
After building the ownCloud Client, install and configure Squish.
```

. Start Squish
. Open the existing test-suite via: menu:File[Open Test Suite > test/gui]
. Go to menu:Edit[Server Settings > Manage AUTs > Mapped AUTs]
. Click btn:[Add] and select the compiled binary, e.g.: `client/client-build/bin/owncloud`
. Close any running Desktop Clients by clicking btn:[Quit ownCloud] in the settings page
. Run the AUT (Application under Test) to check if everything works properly via menu:Run[Lauch AUT] (the Desktop Client settings window should appear)

=== Using Docker

You can also use the {squish-docker-image-url}[Squish docker image] to run tests. Proceed with the following steps:

. Copy `server.ini.` file from `test/gui/drone` to a new folder called `local`
. Change `AUT/owncloud` value to `"/app/client-build/bin"`
. Pull the docker image with the following command:
+
[source,console]
----
sudo docker pull owncloudci/squish
----

== Run Tests

* xref:run-tests-using-squish[Run Tests Using Squish]
* xref:run-tests-using-docker[Run Tests Using Docker]

=== Run Tests Using Squish

* Start the {owncloud-test-middleware-url}[owncloud-test-middleware]

> * Click the play-button for a {test-case-scenario-url}[test-case or scenario].

Reviewer: ```suggestion
* Click the play button for a {test-case-scenario-url}[test-case or scenario].
```

=== Run Tests Using Docker

> * Run squish docker image using the following command:

Reviewer: ```suggestion
* Run the Squish docker image using the following command:
```
